src/ifaddrs.rs

```diff
@@ -0,0 +1,180 @@
+//! Query network interface addresses
+//!
+//! Uses the Linux and/or BSD specific function getifaddrs to query the list
+//! of interfaces and their associated addresses.
+
+use std::ffi;
+use std::iter::Iterator;
+use std::mem;
+use std::option::Option;
+
+use libc;
+use {Result, Error};
+
+use sys::socket::InetAddr;
+
+#[cfg(any(target_os = "linux", target_os = "emscripten", target_os = "fuchsia"))]
```

Comment:
Android doesn't support `getifaddrs`, that's probably why I excluded it. But you're right, the flags themselves are known to Android as well.

Maybe I should move `InterfaceFlags` to `net/if_.rs`, as those constants are defined in `/usr/include/net/if.h` and not in `ifaddrs.h`.